File: 재귀/2447/2447.py
import sys

def star(i, j, n):
   if (i // n) % 3 == 1 and (j // n) % 3 == 1:
      return ' '
   elif(n // 3 == 0):
      return '*'
   else:
      return star(i, j, n//3)

n = int(input())
s = [None] * (n + 1) * n
for i in range(n):
   for j in range(n):
      s[i * (n + 1) + j] = star(i, j, n)
   s[i * (n + 1) - 1] = '\n'
sys.stdout.write(''.join(s))

File: 재귀/2447/2447.c
#include <stdio.h>

void star(int, int, int);

int main()
{
   int n, i, j;
   scanf("%d", &n);
   for (i = 0; i < n; ++i)
   {
      for (j = 0; j < n; ++j)
         star(i, j, n);
      printf("\n");
   }

   return 0;
}

void star(int i, int j, int n)
{
   if((i / n) % 3 == 1 && (j / n) % 3 == 1)
      printf(" ");
   else if(n / 3 == 0)
      printf("*");
   else
      star(i, j, n/3);
}